lib/Doctrine/Common/Annotations/DocParser.php

```diff
@@ -1153,7 +1153,17 @@ private function Constant()
             throw AnnotationException::semanticalErrorConstants($identifier, $this->context);
         }
 
-        return constant($identifier);
+		$value = constant($identifier);
+
+		/** @noinspection PhpElementIsNotAvailableInCurrentPhpVersionInspection */
+		if (interface_exists('\UnitEnum') && $value instanceof \UnitEnum) {
+			if (!property_exists($value, 'value')) {
```

Comment:
the right way to check that would be checking for `BackedEnum` instead

Reply:
BackedEnum is not yet implemented natively by backed enums as of php 8.1, so calling/checking instaneocf BackedEnum always returns false even with backed enums unless you implement if manually, check first user contributed note here https://www.php.net/manual/en/class.backedenum.php

Reply:
Apologies, the IntBackedEnum and StringBackedEnum interfaces (which sadly do have stubs in phpstorm) were problematic/not implemented so i mistook it. Should be fixed/improved now

Reply:
IntBackedEnum and StringBackedEnum are PHPStorm inventions in their stub types. Those interfaces don't exist in PHP (and are not planned to exist as they are not proposed at all).